pkg/controller/machinepool/machinepool_controller.go

```diff
@@ -813,12 +814,23 @@ func (r *ReconcileMachinePool) syncClusterAutoscaler(
 		}
 	}
 	if defaultClusterAutoscaler != nil {
-		if spec := &defaultClusterAutoscaler.Spec; spec.ScaleDown == nil || !spec.ScaleDown.Enabled {
-			logger.Info("updaing cluster autoscaler")
-			if spec.ScaleDown == nil {
-				spec.ScaleDown = &autoscalingv1.ScaleDownConfig{}
-			}
+		spec := &defaultClusterAutoscaler.Spec
+		changed := false
+		if spec.ScaleDown == nil {
+			spec.ScaleDown = &autoscalingv1.ScaleDownConfig{}
+		}
+		// Ensure spec.ScaleDown.Enabled = true
+		if !spec.ScaleDown.Enabled {
 			spec.ScaleDown.Enabled = true
+			changed = true
+		}
+		// Ensure spec.BalanceSimilarNodeGroups = true.
+		if spec.BalanceSimilarNodeGroups == nil || !*spec.BalanceSimilarNodeGroups {
```

Comment:
I don't think we should override this if it's already set.
```suggestion
		// Default spec.BalanceSimilarNodeGroups to true if unset.
		if spec.BalanceSimilarNodeGroups == nil {
```